presto-raptor/src/test/java/com/facebook/presto/raptor/systemtables/TestShardMetadataRecordCursor.java

```diff
@@ -95,19 +98,28 @@ public void teardown()
     }
 
     @Test
-    public void testSimple()
+    public void testTemporalColumnDate()
             throws Exception
     {
         ShardManager shardManager = createShardManager(dbi);
 
-        // Add shards to the table
-        long tableId = 1;
+        SchemaTableName tableName = new SchemaTableName("testsimple", "orders");
+        metadata.createTable(SESSION, tableMetadataBuilder(tableName)
+                .property("temporal_column", "orderdate")
+                .column("orderkey", BIGINT)
+                .column("orderdate", DATE)
+                .build());
+
+        MetadataDao metadataDao = dummyHandle.attach(MetadataDao.class);
```

Comment:
Update all these `attach()` calls to use `dbi.onDemand()`. A `Handle` is a JDBC `Connection` and we only use the dummy handle to keep the H2 database from getting garbaged collected early. Database operations should use a new handle/connection via `onDemand()`. Technically, attaching works fine for these tests, but is better to be consistent in usage across the tests.